### ISSUE TITLE
Push down supertypes

### ISSUE THREAD

**Issue by cbizon** [NCATS-Gamma/robokop#110]
Independent of what we do on the query side, we do want to push nodes down to their lowest subtypes for pushing into neo4j

**Comment by cbizon** [comment]
@patrickkwang Thoughts on this scenario?:

GeneticCondition is a subclass of Disease.  Every GC is a Disease, but not vice versa.  If I push everything down to the leaves of the type-tree then every Disease that is a GC will have the GC label.  If I stop at Disease, then none will.

I guess the best way to do this is: any time I go to write a disease, check to see if it's also a GC, and add both labels to the node?  Then any process running later can find the node however it needs to...

**Comment by patrickkwang** [comment]
Yes, that sounds like a good solution. An alternative is to leave off the "disease" label on the genetic conditions. The meaning would be unambiguous, but that puts perhaps undue burden on things querying the graph. So I think both is good.

**Comment by cbizon** [comment]
Yep, I think that this is what it comes down to: do we want things querying the neo4j (like the ranker-related calls) to know about the biolink-model and reason over its classes or not?

I'd say no - keep pulling from the neo4j as simple as possible.
